spec/facility_spec.rb

```diff
@@ -23,4 +30,131 @@
+  
+      
+   
+      @facility.register_vehicle(@bolt)
+      expect(@facility.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@bolt.registration_date).to eq(Date.today.year)
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility.collected_fees).to eq(325)
+       
+    end
+  end
+
+  describe'#check facility and facility_2' do 
+    it 'sees what is in the first facility' do 
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(@cruz)
+      @facility.register_vehicle(@camaro)
+      @facility.register_vehicle(@bolt)
+      expect(@facility.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+    end
+
+    it 'facility 2 default' do
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.services).to eq([])
+      expect(@facility_2.register_vehicle(@bolt)).to eq(nil)
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.collected_fees).to eq(0)
+    end
+  end
+
+  
+  describe '#Written test' do
+    it 'Registrent 1 without adding service written test' do
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(@registrant_1.permit?).to eq(true)
+      expect(@facility.administer_written_test(@registrant_1)).to eq(false)
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
+
+    it 'is administering the written test' do
+      @facility.add_service('Written Test')
+      expect(@facility.administer_written_test(@registrant_1)).to eq(true)
+      expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+     
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to eq(false)
+      expect(@facility.administer_written_test(@registrant_2)).to eq(false)
+      @registrant_2.earn_permit
+      expect(@facility.administer_written_test(@registrant_2)).to eq(true)
+      expect(@registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      
+      expect(@registrant_3.age).to eq(15)
+      expect(@registrant_3.permit?).to eq(false)
+      expect(@facility.administer_written_test(@registrant_3)).to eq(false)
+      expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
+  end
+
+  describe'#Administar road test' do
+    it 'is andministaring the road test' do 
+      expect(@facility.administer_road_test(@registrant_3)).to eq(false)
+      @registrant_3.earn_permit
+      expect(@facility.administer_road_test(@registrant_3)).to eq(false)
+      expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      
+      expect(@facility.administer_road_test(@registrant_1)).to eq(false)
+      @facility.add_service('Written Test')
+      @facility.add_service('Road Test')
+      expect(@facility.administer_road_test(@registrant_1)).to eq(true)
```

Comment:
Actually, this should be false since the registrant_3 does not yet have their written test passed. These tests were a bit complicated because there were so many specific conditions that needed to be met in different orders. YOu were on the right track here, just need to have a process/system to keep the conditions straight. I like to write them all out on paper or draw pictures. :)